internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -202,11 +215,27 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context, clust
 		update = true
 	}
 
-	// TODO: get from AgentEnvSpec
-	//updateString(spec.HTTPProxy, c.HTTPProxy, &params.HTTPProxy)
-	//updateString(spec.HTTPSProxy, c.HTTPSProxy, &params.HTTPSProxy)
-	//updateString(spec.NoProxy, c.NoProxy, &params.NoProxy)
-	//updateString(spec.AdditionalNtpSource, c.AdditionalNtpSource, &params.AdditionalNtpSource)
+	r.Log.Info("XXXXXXXXX: func updateIfNeeded: before getInstallEnvByClusterDeployment")
+	installEnv, err := getInstallEnvByClusterDeployment(ctx, r.Client, cluster)
+	r.Log.Info("XXXXXXXXX: func updateIfNeeded: before if got installEnv") // TODO before merge: remove
+	if err == nil && installEnv != nil {
+		r.Log.Info("XXXXXXXXX: func updateIfNeeded: in if got installEnv") // TODO before merge: remove
+		//if len(installEnv.Spec.SSHAuthorizedKeys) > 0 {
+		//	updateString(installEnv.Spec.SSHAuthorizedKeys[0], c.SSHPublicKey, &params.SSHPublicKey) // TODO before merge: make sure we don't need this and keep line 220
+		//}
+		if len(installEnv.Spec.AdditionalNTPSources) > 0 {
+			updateString(installEnv.Spec.AdditionalNTPSources[0], c.AdditionalNtpSource, &params.AdditionalNtpSource)
+		}
+		r.Log.Infof("XXXXXXXXX: func updateIfNeeded: in if got installEnv: print NoProxy from installEnv Spec: %s", installEnv.Spec.Proxy.NoProxy)       // TODO before merge: remove
+		r.Log.Infof("XXXXXXXXX: func updateIfNeeded: in if got installEnv: print HTTPProxy from installEnv Spec: %s", installEnv.Spec.Proxy.HTTPProxy)   // TODO before merge: remove
+		r.Log.Infof("XXXXXXXXX: func updateIfNeeded: in if got installEnv: print HTTPSProxy from installEnv Spec: %s", installEnv.Spec.Proxy.HTTPSProxy) // TODO before merge: remove
+		updateString(installEnv.Spec.Proxy.NoProxy, c.NoProxy, &params.NoProxy)
```

Comment:
if one of the parameters is updated then you need to call for a reconcile on install Env. otherwise the ISO may be generated without those parameters.

Reply:
if one of those parameters changes in installEnv, it will first get detected by [notifyClusterUpdatesIfNeeded]( https://github.com/openshift/assisted-service/pull/1170/files#diff-d6f17279093c7d93d612ed5b1f45cb8ba9893abd5e0fc7bdf26ac7203f2b934dR68), and trigger this `ClusterDeploymentsReconciler` flow.

ensureISO  will also [invoke GenerateClusterISOInternal](https://github.com/openshift/assisted-service/pull/1170/files#diff-d6f17279093c7d93d612ed5b1f45cb8ba9893abd5e0fc7bdf26ac7203f2b934dR175) the to regenerate image for relevant changes